src/main/java/com/main/repository/MessageRepository.java

```diff
@@ -0,0 +1,21 @@
+package com.main.repository;
+
+import java.util.List;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import com.main.entity.Message;
+
+@Repository
+public interface MessageRepository extends JpaRepository<Message, Long> {
+
+	@Query("SELECT m FROM Message m WHERE m.person.id = :personId AND (:searchWord IS NULL OR m.content LIKE %:searchWord%)")
```

Comment:
If you are using JPA don't use manual queries. Remove this and use JPA automatic custom queries. Only if you are not getting the repository entity you should go for a manual query in my opinion.

Reply:
I'm been having some problem with the creation of JPA without query